evalml/pipelines/pipeline_base.py

```diff
@@ -487,3 +488,14 @@ def __eq__(self, other):
             if getattr(self, attribute) != getattr(other, attribute):
                 return False
         return True
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+
+        def repr_component(parameters):
+            return ', '.join([f"'{key}': {safe_repr(value)}" for key, value in parameters.items()])
+
+        parameters_repr = ' '.join([f"'{component}':{{{repr_component(parameters)}}}," for component, parameters in self.parameters.items()])
+        return f'{(type(self).__name__)}(parameters={{{parameters_repr}}})'
```

Comment:
I believe [you want double curly braces instead of triple](https://stackoverflow.com/questions/5466451/how-can-i-print-literal-curly-brace-characters-in-python-string-and-also-use-fo) here.

Reply:
Since the parameters is a dictionary and contains an expression, I do need all three! The first two are for literal curly braces and the third is for the formatting.

Reply:
Hm interesting, but then the output (taken from the unit test) ends up as
```python
MockPipeline(parameters={{'Imputer':{{'categorical_impute_strategy': 'most_frequent', 'numeric_impute_strategy': 'mean', 'categorical_fill_value': None, 'numeric_fill_value': None}}, '{final_estimator}':{{'n_estimators': 100, 'max_depth': 6, 'n_jobs': -1}},}})
```

But ideally I think it should be
```python
MockPipeline(parameters={'Imputer': {'categorical_impute_strategy': 'most_frequent', 'numeric_impute_strategy': 'mean', 'categorical_fill_value': None, 'numeric_fill_value': None}, 'final_estimator': {'n_estimators': 100, 'max_depth': 6, 'n_jobs': -1}})
```

Do you agree? The second output could be evaluated in the python repl and turned into an object. I think the first would fail.

Reply:
@eccabay but, I think its fine to merge this and we can circle back. It adds value even if there's a couple details we may wanna discuss further :)

Reply:
The double curly braces in the unit tests are once again for f-string formatting! If you actually print the repr, you get the second code block you pasted there, and calling eval on the expected_repr produces the correct object.

Reply:
@eccabay hmm could you paste an example? I don't follow yet. I thought that `__repr__` returns a string, which requires no further formatting, and that's the end of it 😂  In other words, I thought we should output the second snippet, because that can be pasted into the python REPL and evaluated, but we're currently outputting the first.

Again, not blocking merge, I'm just trying to make sure we all agree on desired behavior.

Reply:
@eccabay ok, I think I understand now. So the string I copied was a format string, which is why you need the double curlys there.

I just did a test locally and everything looks great

![Screen Shot 2020-09-29 at 10 40 39 AM](https://user-images.githubusercontent.com/235870/94573939-95d72700-0240-11eb-8062-1bebadee787e.png)

Realizing the unit tests were using a format string was the key point I was missing, lol. Thanks! 😆